node/impl/full/actor_events.go

```diff
@@ -39,188 +40,320 @@ type ActorEventsAPI struct {
 	ActorEventAPI
 }
 
-func (a *ActorEventHandler) GetActorEvents(ctx context.Context, filter *types.ActorEventFilter) ([]*types.ActorEvent, error) {
+func (a *ActorEventHandler) GetActorEvents(ctx context.Context, evtFilter *types.ActorEventFilter) ([]*types.ActorEvent, error) {
 	if a.EventFilterManager == nil {
 		return nil, api.ErrNotSupported
 	}
 
-	params, err := a.parseFilter(filter)
+	if evtFilter == nil {
+		evtFilter = &types.ActorEventFilter{}
+	}
+	params, err := a.parseFilter(*evtFilter)
 	if err != nil {
 		return nil, err
 	}
 
-	// Create a temporary filter
-	f, err := a.EventFilterManager.Install(ctx, params.MinHeight, params.MaxHeight, params.TipSetCid, filter.Addresses, filter.Fields, false)
+	// Install a filter just for this call, collect events, remove the filter
+
+	tipSetCid, err := params.GetTipSetCid()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get tipset cid: %w", err)
+	}
+	f, err := a.EventFilterManager.Install(ctx, params.MinHeight, params.MaxHeight, tipSetCid, evtFilter.Addresses, evtFilter.Fields, false)
 	if err != nil {
 		return nil, err
 	}
 
-	evs, err := getCollected(ctx, f)
-	_ = a.EventFilterManager.Remove(ctx, f.ID())
-	return evs, err
+	evs, _, _ := getCollected(ctx, f)
+	if err := a.EventFilterManager.Remove(ctx, f.ID()); err != nil {
+		log.Warnf("failed to remove filter: %s", err)
+	}
+	return evs, nil
 }
 
 type filterParams struct {
 	MinHeight abi.ChainEpoch
 	MaxHeight abi.ChainEpoch
-	TipSetCid cid.Cid
+	TipSetKey types.TipSetKey
+}
+
+func (fp filterParams) GetTipSetCid() (cid.Cid, error) {
+	if fp.TipSetKey.IsEmpty() {
+		return cid.Undef, nil
+	}
+	return fp.TipSetKey.Cid()
 }
 
-func (a *ActorEventHandler) parseFilter(f *types.ActorEventFilter) (*filterParams, error) {
-	if f.TipSetCid != nil {
-		if len(f.FromEpoch) != 0 || len(f.ToEpoch) != 0 {
-			return nil, fmt.Errorf("cannot specify both TipSetCid and FromEpoch/ToEpoch")
+func (a *ActorEventHandler) parseFilter(f types.ActorEventFilter) (*filterParams, error) {
+	if f.TipSetKey != nil && !f.TipSetKey.IsEmpty() {
+		if f.FromHeight != nil || f.ToHeight != nil {
+			return nil, fmt.Errorf("cannot specify both TipSetKey and FromHeight/ToHeight")
 		}
 
+		tsk := types.EmptyTSK
+		if f.TipSetKey != nil {
```

Comment:
redundant

Reply:
I won't admit how long I looked at this wondering what Go magic I was missing .. before noticing the enclosing `if`